compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -345,18 +408,18 @@ static void buildReductionCudaStrategy(ImplicitLocOpBuilder &b, Value variantH,
   // Step 0. Match the ops.
   Value originalFillH =
       b.create<MatchOp>(variantH, linalg::FillOp::getOperationName());
-  auto [reductionH, fusionRootH] =
-      reductionBlockDistributionHandles(b, variantH, infos.hasTrailingEltwise);
+  auto [leadingH, reductionH, fusionRootH] = reductionBlockDistributionHandles(
```

Comment:
1st commit:`FailureOr<Value>` and try to avoid propagating magic API bools ?
Here and below.